### DISCUSSION TITLE
Is it possible to build extensions without using npx?

### DISCUSSION THREAD

**Discussion by infomiho** [directus/directus#7977] [Archive]
Hey everyone, I'm developing a custom interface and here is my current setup with `npx`.

```
{
    "scripts": {
        "build": "npx @directus/extensions-sdk build -o index.js"
    },
    "directus:extension": {
        "type": "interface",
        "path": "index.js",
        "source": "src/index.js",
        "host": "^9.0.0-rc.92",
        "hidden": false
    },
    "devDependencies": {
        "@directus/extensions-sdk": "^9.0.0-rc.92",
    },
}
```

The main downside of the current approach is that `npx` reinstalls its dependencies EACH time. Which takes a lot of time if you keep rebuilding the extension 😃

I've tried setting up a build command which relies on already installed dev dependency but I can't seem to get it to work. Does anyone have a suggestion how to build extensions without using `npx`?

**Comment by rijkvanzanten** [comment]
That doesnt sound right to me 🤔 If the dependency exists in the local node modules folder, npx should use the local one instead of installing it every time. Did you run npm install before?

**Comment by nickrum** [comment]
When a package is installed, you can just call it's binaries in `scripts` directly without having to use `npx`:

```js
"scripts": {
  "build": "directus-extension build"
}
```

**Reply by infomiho** [reply]
@nickrum Hey man, I've tried it and this is what happens:
```
    "scripts": {
        "build": "extensions-sdk build -o index.js"
    },
```

```
❯ npm run build

> picker@1.0.0 build /Users/ilakovac/directus-extensions/extensions/interfaces/picker
> extensions-sdk build -o index.js

sh: extensions-sdk: command not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! picker@1.0.0 build: `extensions-sdk build -o index.js`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the picker@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/ilakovac/.npm/_logs/2021-09-13T09_49_37_234Z-debug.log
```

**Reply by nickrum** [reply]
`extensions-sdk` is the name of the package, the CLI binary is called `directus-extension`. You have to use `directus-extension build`.

**Reply by infomiho** [reply]
@nickrum Sorry for missing that little piece of difference. It would be great to have a section in the docs dedicated to extensions dev workflow 😄 

```
  "scripts": {
      "build": "directus-extension build -o index.js"
  },
```

This setup works for me now! Thanks for the help.

**Reply by nickrum** [reply]
No worries, I can see how this might be confusing! We are already working on improving the extension docs 👍
